Remotion/Data/DomainObjects/Infrastructure/ObjectPersistence/RootPersistenceStrategy.cs

```diff
@@ -139,7 +146,8 @@ public virtual IEnumerable<IQueryResultRow> ExecuteCustomQuery (IQuery query)
       if (query.QueryType != QueryType.CustomReadOnly && query.QueryType != QueryType.CustomReadWrite)
         throw new ArgumentException("Only custom queries can be used to load custom results", "query");
 
-      using (var storageProviderManager = CreateStorageProviderManager())
+      var storageAccess = query.QueryType.IsReadWrite() ? StorageAccessType.ReadWrite : _storageAccessResolver.ResolveStorageAccessForQuery(query);
```

Comment:
We can (should?) hide this line in a helper method and reuse in the other lines, too. This method can actually cover the IsReadWrite semantics instead of having a separate helper method.